My develop

njazz opened the pull request:
cpd_system_list_bind_objects() cpd_system_list_loaded_classes() names
fixed

njazz committed:
include sys/types.h

njazz committed:
loadLibrary()

njazz committed:
console stub

njazz committed:
build fix

njazz committed:
try-catch for new object

njazz committed:
createObject arguments parsing stub

njazz committed:
bind object to symbol

njazz committed:
cpd_set_verbose_level() added

njazz committed:
PdLocalProcess::setLogLevel() added

njazz committed:
Merge remote-tracking branch 'uliss/devel' into my-develop

njazz committed:
fixes

njazz committed:
pd console observer stub

njazz committed:
fix for previous commit

njazz committed:
cpd_disconnect() added

njazz committed:
PdCanvas disconnect() added

njazz committed:
disconnect() fix

njazz committed:
add xlets to pd objects

njazz committed:
prototypes for cmp_system_list_bind_objects(),
cmp_system_list_loaded_classes()

njazz committed:
dspSwitch()

njazz committed:
PdLocalProcess::loadedClasses() stub
cpd_system_list_bind_objects() cpd_system_list_loaded_classes() names
fixed

njazz committed:
Merge remote-tracking branch 'uliss/devel' into my-develop

njazz committed:
Merge remote-tracking branch 'uliss/devel' into my-develop

njazz committed:
cpd_receiver

njazz committed:
PdObjectObserver callback action

njazz committed:
fixes for previous commit

njazz committed:
fix

njazz committed:
more fixes

njazz committed:
fixes

njazz committed:
Merge remote-tracking branch 'uliss/devel' into my-develop

njazz committed:
fixed receiverCallback()

uliss commented on pd/cpd/cpd_receiver.cpp:
```diff
@@ -0,0 +1,64 @@
+
+static void cpd_receiver_any(t_receiver* x, t_symbol* sel, int argc, t_atom* argv)
+{
+    t_cpd_list* msg = cpd_list_new(0);
```
list allocated, but not freed

uliss commented on pd/cpd/cpd_receiver.cpp:
```diff
@@ -0,0 +1,64 @@
+        cpd_list_append(msg, &argv[i]);
+
+    //callback
+    if (x->callback)
```
если колбек не установлен - нужно сразу возвращаться, без копирования аргументов

uliss commented on pd/cpd/cpd_receiver.h:
```diff
@@ -0,0 +1,23 @@
+typedef struct _receiver t_receiver;
+typedef void (*t_cpd_receiver_callback)(t_cpd_list* msg);
+
+CPD_EXTERN void cpd_receiver_init();
```
возможно стоит скрыть этот вызов, а вызывать его  в cpd_init()

njazz replied:
ок, я тогда добавлю сразу - не хотел править его

uliss commented on src/pd_localprocess.cpp:
```diff
@@ -73,4 +88,93 @@ void PdLocalProcess::unregisterConsoleObserver(ConsoleObserverPtr o)
+        char b[64];
+        sprintf(b, "%lu", (long)lPtr);
+
+        xpd::log()->info(b);
```
в лог можно выводить так: xpd::log()->info("message: {}, {}, {}", 1, 2, 3) 
без буфера

uliss commented on src/pd_localprocess.cpp:
```diff
@@ -73,4 +88,93 @@ void PdLocalProcess::unregisterConsoleObserver(ConsoleObserverPtr o)
+
+        long lPtr;
+        stream >> lPtr;
+
```
не очень понятно зачем такая цепочка: 
t_cpd_list* -> t_cpd_symbol* -> const char* -> std::stringstream -> long -> t_cpd_object*

njazz replied:
а вот тут можно подумать - я делал на стороне UI объектов так: указатель на объект -> long -> строку через sstream ну и далее строка в символ
можно заменить на Data

uliss commented on src/pd_localprocess.cpp:
```diff
@@ -73,4 +88,93 @@ void PdLocalProcess::unregisterConsoleObserver(ConsoleObserverPtr o)
+        if (!PdLocalProcess::objectObserverMap[objPtr])
+            return;
+
+        ObserverPtr p = PdLocalProcess::objectObserverMap[objPtr];
```
кажется этот код всегда возвращает ссылку, даже если объект не найдет он вставляет его:
http://www.cplusplus.com/reference/map/map/operator[]/

uliss replied:
искать элемент в мэпе нужно через map::find() и проверкой c  map::end()

njazz replied:
да, заменю, точно.

uliss commented on src/pd_localprocess.cpp:
```diff
@@ -1,15 +1,29 @@
+    cpd_bind_object(reinterpret_cast<t_cpd_object*>(receiver_), cpd_symbol("xpd_receiver"));
 }
 
 PdLocalProcess::~PdLocalProcess()
```
receiver_ не удаляется в деструкторе

njazz replied:
ага, я опять забыл)

uliss reviewed (changes requested):
Леш, подправь, плиз, немного, что прокомментировал

njazz committed:
fixes

njazz committed:
test fix
(check that later)

njazz committed:
PdArguments(Arguments) constructor added

njazz committed:
fix 1

njazz committed:
fix 2

njazz committed:
working PdObject::sendList()

njazz committed:
added PdObjects::sendStringAsList(std::string s)
uses cpd_list_new_from_string to parse string to atom list using native
Pd binbuf

njazz committed:
synced from other repository

njazz committed:
fixes
